utils/api/search.ts

```diff
@@ -11,28 +38,26 @@ export const fetchSearch = (search: string) => {
       console.warn(err)
     }
   }
-
-  return fetch(`${API_ENDPOINT}/search?keyword=${query}`)
+  return fetchSearchResult({ text: query })
     .then(async res => {
-      if (res.status === HttpStatus.NotFound) {
+      if (!res || !res.id) {
         return `/404`
       }
-      const found: Record<'id' | 'type', string> | ErrorResponse = await res.json()
-      if (isError(found)) {
-        return `/404`
-      }
-      switch (found.type) {
-        case 'block': {
-          return `/block/${found.id}`
+      switch (res.type) {
+        case 'BLOCK': {
+          return `/block/${res.id}`
+        }
+        case 'TRANSACTION': {
+          return `/tx/${res.id}`
         }
-        case 'transaction': {
-          return `/tx/${found.id}`
+        case 'ACCOUNT': {
+          return `/account/${res.id}`
         }
-        case 'account': {
-          return `/account/${found.id}`
+        case 'ADDRESS': {
```

Comment:
for Address type, do we need to redirect to account page or 404 page? since the page doesn't have content, should we redirect to 404? @Keith-CY

Reply:
The difference between `Account` and `Address` is that `Account` has sent transactions on godwoken, so it's initialized on godwoken. While `Address` has not sent a transaction so it's not initialized on godwoken.

But it's possible to send tokens to an `Address`, then `Address` will hold assets. It should be allowed to view assets held by an `Address` so the page should be accessible.